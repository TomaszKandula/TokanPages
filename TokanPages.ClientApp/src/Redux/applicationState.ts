import { IArticle } from "./States/Articles/selectArticleState";
import { IArticles } from "./States/Articles/listArticlesState";
import { IUpdateArticle } from "./States/Articles/updateArticleState";
import { IAddSubscriber } from "./States/Subscribers/addSubscriberState";
import { IUpdateSubscriber } from "./States/Subscribers/updateSubscriberState";
import { IRemoveSubscriber } from "./States/Subscribers/removeSubscriberState";
import { IUpdateUserData } from "./States/Users/updateUserDataState";
import { ISigninUser } from "./States/Users/signinUserState";
import { ISendMessage } from "./States/Mailer/sendMessageState";
import { IRaiseError } from "./States/raiseErrorState";
import { IRaiseDialog } from "./States/raiseDialogState";
import { IGetStaticContent } from "./States/Content/getStaticContentState";
import { IGetArticleFeatContent } from "./States/Content/getArticleFeatContentState";
import { IGetContactFormContent } from "./States/Content/getContactFormContentState";
import { IGetCookiesPromptContent } from "./States/Content/getCookiesPromptContentState";
import { IGetFeaturedContent } from "./States/Content/getFeaturedContentState";
import { IGetFeaturesContent } from "./States/Content/getFeaturesContentState";
import { IGetFooterContent } from "./States/Content/getFooterContentState";
import { IGetHeaderContent } from "./States/Content/getHeaderContentState";
import { IGetNavigationContent } from "./States/Content/getNavigationContentState";
import { IGetNewsletterContent } from "./States/Content/getNewsletterContentState";
import { IGetWrongPagePromptContent } from "./States/Content/getWrongPagePromptContentState";
import { IGetResetPasswordContent } from "./States/Content/getResetPasswordContentState";
import { IGetUserSigninContent } from "./States/Content/getUserSigninContentState";
import { IGetUserSignoutContent } from "./States/Content/getUserSignoutContentState";
import { IGetUserSignupContent } from "./States/Content/getUserSignupContentState";
import { IGetTestimonialsContent } from "./States/Content/getTestimonialsContentState";
import { IGetUnsubscribeContent } from "./States/Content/getUnsubscribeContentState";
import { IGetUpdateSubscriberContent } from "./States/Content/getUpdateSubscriberContentState";

export interface IApplicationState 
{
    raiseError: IRaiseError;
    raiseDialog: IRaiseDialog;
    selectArticle: IArticle;
    listArticles: IArticles;
    updateArticle: IUpdateArticle,
    sendMessage: ISendMessage,
    addSubscriber: IAddSubscriber,
    updateSubscriber: IUpdateSubscriber,
    removeSubscriber: IRemoveSubscriber,
    updateUserData: IUpdateUserData,
    signinUser: ISigninUser,
    getStaticContent: IGetStaticContent,
    getArticleFeatContent: IGetArticleFeatContent,
    getContactFormContent: IGetContactFormContent,
    getCookiesPromptContent: IGetCookiesPromptContent,
    getFeaturedContent: IGetFeaturedContent,
    getFeaturesContent: IGetFeaturesContent,
    getFooterContent: IGetFooterContent,
    getHeaderContent: IGetHeaderContent,
    getNavigationContent: IGetNavigationContent,
    getNewsletterContent: IGetNewsletterContent,
    getWrongPagePromptContent: IGetWrongPagePromptContent,
    getResetPasswordContent: IGetResetPasswordContent,
    getUserSigninContent: IGetUserSigninContent,
    getUserSignoutContent: IGetUserSignoutContent,
    getUserSignupContent: IGetUserSignupContent,
    getTestimonialsContent: IGetTestimonialsContent,
    getUnsubscribeContent: IGetUnsubscribeContent,
    getUpdateSubscriberContent: IGetUpdateSubscriberContent
}

export interface AppThunkAction<TAction> 
{
    (dispatch: (action: TAction) => void, getState: () => IApplicationState): void;
}
