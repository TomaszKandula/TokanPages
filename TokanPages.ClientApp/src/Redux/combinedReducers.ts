import { ActivateAccountReducer } from "./Reducers/Users/activateAccountReducer";
import { AddSubscriberReducer } from "./Reducers/Subscribers/addSubscriberReducer";
import { GetAccountContentReducer } from "./Reducers/Content/getAccountContenReducer";
import { GetActivateAccountContentReducer } from "./Reducers/Content/getActivateAccountContentReducer";
import { GetArticleFeaturesContentReducer } from "./Reducers/Content/getArticleFeaturesContentReducer";
import { GetClientsContentReducer } from "./Reducers/Content/getClientsContentReducer";
import { GetContactFormContentReducer } from "./Reducers/Content/getContactFormContentReducer";
import { GetCookiesPromptContentReducer } from "./Reducers/Content/getCookiesPromptContentReducer";
import { GetFeaturedContentReducer } from "./Reducers/Content/getFeaturedContentReducer";
import { GetFeaturesContentReducer } from "./Reducers/Content/getFeaturesContentReducer";
import { GetFooterContentReducer } from "./Reducers/Content/getFooterContentReducer";
import { GetHeaderContentReducer } from "./Reducers/Content/getHeaderContentReducer";
import { GetNavigationContentReducer } from "./Reducers/Content/getNavigationContentReducer";
import { GetNewsletterContentReducer } from "./Reducers/Content/getNewsletterContentReducer";
import { GetResetPasswordContentReducer } from "./Reducers/Content/getResetPasswordContentReducer";

import { GetTestimonialsContentReducer } from "./Reducers/Content/getTestimonialsContentReducer";
import { GetUnsubscribeContentReducer } from "./Reducers/Content/getUnsubscribeContentReducer";
import { GetUpdatePasswordContentReducer } from "./Reducers/Content/getUpdatePasswordContentReducer";
import { GetUpdateSubscriberContentReducer } from "./Reducers/Content/getUpdateSubscriberContentReducer";
import { GetUserSigninContentReducer } from "./Reducers/Content/getUserSigninContentReducer";
import { GetUserSignoutContentReducer } from "./Reducers/Content/getUserSignoutContentReducer";
import { GetUserSignupContentReducer } from "./Reducers/Content/getUserSignupContentReducer";
import { GetWrongPagePromptContentReducer } from "./Reducers/Content/getWrongPagePromptContentReducer";
import { ListArticlesReducer } from "./Reducers/Articles/listArticlesReducer";
import { RaiseDialogReducer } from "./Reducers/raiseDialogReducer";
import { RaiseErrorReducer } from "./Reducers/raiseErrorReducer";
import { ReAuthenticateUserReducer } from "./Reducers/Users/reAuthenticateUserReducer";
import { RemoveAccountReducer } from "./Reducers/Users/removeAccountReducer";
import { RemoveSubscriberReducer } from "./Reducers/Subscribers/removeSubscriberReducer";
import { ResetUserPasswordReducer } from "./Reducers/Users/resetUserPasswordReducer";
import { SelectArticleReducer } from "./Reducers/Articles/selectArticleReducer";
import { SendMessageReducer } from "./Reducers/Mailer/sendMessageReducer";
import { SigninUserReducer } from "./Reducers/Users/signinUserReducer";
import { SignupUserReducer } from "./Reducers/Users/signupUserReducer";
import { StoreUserDataReducer } from "./Reducers/Users/storeUserDataReducer";
import { UpdateArticleReducer } from "./Reducers/Articles/updateArticleReducer";
import { UpdateSubscriberReducer } from "./Reducers/Subscribers/updateSubscriberReducer";
import { UpdateUserPasswordReducer } from "./Reducers/Users/updateUserPasswordReducer";
import { UpdateUserReducer } from "./Reducers/Users/updateUserReducer";
import { UserLanguageReducer } from "./Reducers/userLanguageReducer";

export const combinedReducers = 
{
    raiseError: RaiseErrorReducer,
    raiseDialog: RaiseDialogReducer,

    userLanguage: UserLanguageReducer,

    listArticles: ListArticlesReducer,
    selectArticle: SelectArticleReducer,
    storeUserData: StoreUserDataReducer,

    sendMessage: SendMessageReducer,
    updateArticle: UpdateArticleReducer,
    addSubscriber: AddSubscriberReducer,
    updateSubscriber: UpdateSubscriberReducer,
    removeSubscriber: RemoveSubscriberReducer,
    removeAccount: RemoveAccountReducer,
    activateAccount: ActivateAccountReducer,
    reAuthenticateUser: ReAuthenticateUserReducer,
    signinUser: SigninUserReducer,
    signupUser: SignupUserReducer,
    updateUser: UpdateUserReducer,
    updateUserPassword: UpdateUserPasswordReducer,
    resetUserPassword: ResetUserPasswordReducer,

    getNavigationContent: GetNavigationContentReducer,
    getHeaderContent: GetHeaderContentReducer,
    getFooterContent: GetFooterContentReducer,

    getAccountContent: GetAccountContentReducer, 
    getActivateAccountContent: GetActivateAccountContentReducer,

    getClientsContent: GetClientsContentReducer,
    getFeaturesContent: GetFeaturesContentReducer,
    getArticleFeaturesContent: GetArticleFeaturesContentReducer,
    getFeaturedContent: GetFeaturedContentReducer,
    getTestimonialsContent: GetTestimonialsContentReducer,
    getNewsletterContent: GetNewsletterContentReducer,
    getContactFormContent: GetContactFormContentReducer,
    getCookiesPromptContent: GetCookiesPromptContentReducer,
    getWrongPagePromptContent: GetWrongPagePromptContentReducer,

    getUserSigninContent: GetUserSigninContentReducer,
    getUserSignupContent: GetUserSignupContentReducer,
    getUserSignoutContent: GetUserSignoutContentReducer,
    getUpdatePasswordContent: GetUpdatePasswordContentReducer,
    getResetPasswordContent: GetResetPasswordContentReducer,

    getUnsubscribeContent: GetUnsubscribeContentReducer,
    getUpdateSubscriberContent: GetUpdateSubscriberContentReducer
};
