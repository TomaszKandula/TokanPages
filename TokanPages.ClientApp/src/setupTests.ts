// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
