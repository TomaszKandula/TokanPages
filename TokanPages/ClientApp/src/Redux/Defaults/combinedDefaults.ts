import { SelectArticleDefault } from "./selectArticleDefault";
import { ListArticlesDefault } from "./listArticlesDefault";
import { UpdateArticleDefault } from "./updateArticleDefault";
import { SendMessageStateDefault } from "./sendMessageDefault";
import { AddSubscriberDefault } from "./addSubscriberDefault";
import { UpdateSubscriberDefault } from "./updateSubscriberDefault";
import { RemoveSubscriberDefault } from "./removeSubscriberDefault";

export const combinedDefaults = 
{
    selectArticle: SelectArticleDefault,
    listArticles: ListArticlesDefault,
    updateArticle: UpdateArticleDefault,
    sendMessage: SendMessageStateDefault,
    addSubscriber: AddSubscriberDefault,
    updateSubscriber: UpdateSubscriberDefault,
    removeSubscriber: RemoveSubscriberDefault
};
