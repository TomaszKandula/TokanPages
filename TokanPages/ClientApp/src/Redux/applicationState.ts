import { IArticle } from "./States/selectArticleState";
import { IArticles } from "./States/listArticlesState";
import { ISendMessage } from "./States/sendMessageState";
import { IAddSubscriber } from "./States/addSubscriberState";
import { IUpdateSubscriber } from "./States/updateSubscriberState";
import { IRemoveSubscriber } from "./States/removeSubscriberState";
import { IUpdateArticle } from "./States/updateArticleState";

export interface IApplicationState 
{
    selectArticle: IArticle;
    listArticles: IArticles;
    updateArticle: IUpdateArticle,
    sendMessage: ISendMessage,
    addSubscriber: IAddSubscriber,
    updateSubscriber: IUpdateSubscriber,
    removeSubscriber: IRemoveSubscriber
}

export interface AppThunkAction<TAction> 
{
    (dispatch: (action: TAction) => void, getState: () => IApplicationState): void;
}
