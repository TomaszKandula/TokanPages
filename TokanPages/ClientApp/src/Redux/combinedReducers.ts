import SelectArticlesReducer from "./Reducers/selectArticleReducer";
import ListArticlesReducer from "./Reducers/listArticlesReducer";
import SendMessageReducer from "./Reducers/sendMessageReducer";
import AddSubscriberReducer from "./Reducers/addSubscriberReducer";
import UpdateSubscriberReducer from "./Reducers/updateSubscriberReducer";
import RemoveSubscriberReducer from "./Reducers/removeSubscriberReducer";
import UpdateArticleReducer from "./Reducers/updateArticleReducer";
import RaiseErrorReducer from "./Reducers/raiseErrorReducer";
import RaiseDialogReducer from "./Reducers/raiseDialogReducer";
import GetStaticContentReducer from "./Reducers/getStaticContentReducer";
import GetArticleFeatContentReducer from "./Reducers/getArticleFeatContentReducer";
import GetContactFormContentReducer from "./Reducers/getContactFormContentReducer";
import GetCookiesPromptContentReducer from "./Reducers/getCookiesPromptContentReducer";
import GetFeaturedContentReducer from "./Reducers/getFeaturedContentReducer";
import GetFeaturesContentReducer from "./Reducers/getFeaturesContentReducer";
import GetFooterContentReducer from "./Reducers/getFooterContentReducer";
import GetHeaderContentReducer from "./Reducers/getHeaderContentReducer";
import GetNavigationContentReducer from "./Reducers/getNavigationContentReducer";
import GetNewsletterContentReducer from "./Reducers/getNewsletterContentReducer";
import GetWrongPagePromptContentReducer from "./Reducers/getWrongPagePromptContentReducer";
import GetResetFormContentReducer from "./Reducers/getResetFormContentReducer";
import GetSigninFormContentReducer from "./Reducers/getSigninFormContentReducer";
import GetSignupFormContentReducer from "./Reducers/getSignupFormContentReducer";
import GetTestimonialsContentReducer from "./Reducers/getTestimonialsContentReducer";
import GetUnsubscribeContentReducer from "./Reducers/getUnsubscribeContentReducer";
import GetUpdateSubscriberContentReducer from "./Reducers/getUpdateSubscriberContentReducer";

export const combinedReducers = 
{
    raiseError: RaiseErrorReducer,
    raiseDialog: RaiseDialogReducer,
    selectArticle: SelectArticlesReducer,
    listArticles: ListArticlesReducer,
    updateArticle: UpdateArticleReducer,
    sendMessage: SendMessageReducer,
    addSubscriber: AddSubscriberReducer,
    updateSubscriber: UpdateSubscriberReducer,
    removeSubscriber: RemoveSubscriberReducer,
    getStaticContent: GetStaticContentReducer,
    getArticleFeatContent: GetArticleFeatContentReducer,
    getContactFormContent: GetContactFormContentReducer,
    getCookiesPromptContent: GetCookiesPromptContentReducer,
    getFeaturedContent: GetFeaturedContentReducer,
    getFeaturesContent: GetFeaturesContentReducer,
    getFooterContent: GetFooterContentReducer,
    getHeaderContent: GetHeaderContentReducer,
    getNavigationContent: GetNavigationContentReducer,
    getNewsletterContent: GetNewsletterContentReducer,
    getWrongPagePromptContent: GetWrongPagePromptContentReducer,
    getResetFormContent: GetResetFormContentReducer,
    getSigninFormContent: GetSigninFormContentReducer,
    getSignupFormContent: GetSignupFormContentReducer,
    getTestimonialsContent: GetTestimonialsContentReducer,
    getUnsubscribeContent: GetUnsubscribeContentReducer,
    getUpdateSubscriberContent: GetUpdateSubscriberContentReducer
};
