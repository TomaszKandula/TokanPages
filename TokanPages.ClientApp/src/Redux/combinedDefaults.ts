import { IApplicationState } from "./applicationState";
import { SelectArticleDefault } from "./Defaults/selectArticleDefault";
import { ListArticlesDefault } from "./Defaults/listArticlesDefault";
import { UpdateArticleDefault } from "./Defaults/updateArticleDefault";
import { SendMessageStateDefault } from "./Defaults/sendMessageDefault";
import { AddSubscriberDefault } from "./Defaults/addSubscriberDefault";
import { UpdateSubscriberDefault } from "./Defaults/updateSubscriberDefault";
import { RemoveSubscriberDefault } from "./Defaults/removeSubscriberDefault";
import { RaiseErrorDefault } from "./Defaults/raiseErrorDefault";
import { RaiseDialogDefault } from "./Defaults/raiseDialogDefault";
import { UpdateUserDataDefault } from "./Defaults/updateUserDataDefault";
import { SigninUserDefault } from "./Defaults/signinUserDefault";
import { GetStaticContentDefault } from "./Defaults/getStaticContentDefault";
import { GetArticleFeatContentDefault } from "./Defaults/getArticleFeatContentDefault";
import { GetContactFormContentDefault } from "./Defaults/getContactFormContentDefault";
import { GetCookiesPromptContentDefault } from "./Defaults/getCookiesPromptContentDefault";
import { GetFeaturedContentDefault } from "./Defaults/getFeaturedContentDefault";
import { GetFeaturesContentDefault } from "./Defaults/getFeaturesContentDefault";
import { GetFooterContentDefault } from "./Defaults/getFooterContentDefault";
import { GetHeaderContentDefault } from "./Defaults/getHeaderContentDefault";
import { GetNavigationContentDefault } from "./Defaults/getNavigationContentDefault";
import { GetNewsletterContentDefault } from "./Defaults/getNewsletterContentDefault";
import { GetWrongPagePromptContentDefault } from "./Defaults/getWrongPagePromptContentDefault";
import { GetResetFormContentDefault } from "./Defaults/getResetFormContentDefault";
import { GetSigninFormContentDefault } from "./Defaults/getSigninFormContentDefault";
import { GetSignupFormContentDefault } from "./Defaults/getSignupFormContentDefault";
import { GetTestimonialsContentDefault } from "./Defaults/getTestimonialsContentDefault";
import { GetUnsubscribeContentDefault } from "./Defaults/getUnsubscribeContentDefault";
import { GetUpdateSubscriberContentDefault } from "./Defaults/getUpdateSubscriberContentDefault";

export const combinedDefaults: IApplicationState = 
{
    raiseError: RaiseErrorDefault,
    raiseDialog: RaiseDialogDefault,
    selectArticle: SelectArticleDefault,
    listArticles: ListArticlesDefault,
    updateArticle: UpdateArticleDefault,
    sendMessage: SendMessageStateDefault,
    addSubscriber: AddSubscriberDefault,
    updateSubscriber: UpdateSubscriberDefault,
    removeSubscriber: RemoveSubscriberDefault,
    updateUserData: UpdateUserDataDefault,
    signinUser: SigninUserDefault,
    getStaticContent: GetStaticContentDefault,
    getArticleFeatContent: GetArticleFeatContentDefault,
    getContactFormContent: GetContactFormContentDefault,
    getCookiesPromptContent: GetCookiesPromptContentDefault,
    getFeaturedContent: GetFeaturedContentDefault,
    getFeaturesContent: GetFeaturesContentDefault,
    getFooterContent: GetFooterContentDefault,
    getHeaderContent: GetHeaderContentDefault,
    getNavigationContent: GetNavigationContentDefault,
    getNewsletterContent: GetNewsletterContentDefault,
    getWrongPagePromptContent: GetWrongPagePromptContentDefault,
    getResetFormContent: GetResetFormContentDefault,
    getSigninFormContent: GetSigninFormContentDefault,
    getSignupFormContent: GetSignupFormContentDefault,
    getTestimonialsContent: GetTestimonialsContentDefault,
    getUnsubscribeContent: GetUnsubscribeContentDefault,
    getUpdateSubscriberContent: GetUpdateSubscriberContentDefault
};
