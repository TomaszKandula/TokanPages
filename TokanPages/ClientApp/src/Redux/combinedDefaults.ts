import { SelectArticleDefault } from "./Defaults/selectArticleDefault";
import { ListArticlesDefault } from "./Defaults/listArticlesDefault";
import { UpdateArticleDefault } from "./Defaults/updateArticleDefault";
import { SendMessageStateDefault } from "./Defaults/sendMessageDefault";
import { AddSubscriberDefault } from "./Defaults/addSubscriberDefault";
import { UpdateSubscriberDefault } from "./Defaults/updateSubscriberDefault";
import { RemoveSubscriberDefault } from "./Defaults/removeSubscriberDefault";
import { RaiseErrorDefault } from "./Defaults/raiseErrorDefault";
import { GetStaticContentDefault } from "./Defaults/getStaticContentDefault";

export const combinedDefaults = 
{
    raiseError: RaiseErrorDefault,
    selectArticle: SelectArticleDefault,
    listArticles: ListArticlesDefault,
    updateArticle: UpdateArticleDefault,
    sendMessage: SendMessageStateDefault,
    addSubscriber: AddSubscriberDefault,
    updateSubscriber: UpdateSubscriberDefault,
    removeSubscriber: RemoveSubscriberDefault,
    getStaticContent: GetStaticContentDefault
};
