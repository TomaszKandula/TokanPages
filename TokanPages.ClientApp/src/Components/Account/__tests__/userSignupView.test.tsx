import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import SignupFormView from "../userSignupView";
import { combinedDefaults } from "../../../Redux/combinedDefaults";

describe("Test account component: SignupForm.", () => 
{
    it("Renders correctly '<SignupFormView />' when content is loaded.", () => 
    {
        const tree = shallow(<SignupFormView content={combinedDefaults.getSignupFormContent.content} isLoading={false} />);
        expect(tree).toMatchSnapshot();
    });
});
