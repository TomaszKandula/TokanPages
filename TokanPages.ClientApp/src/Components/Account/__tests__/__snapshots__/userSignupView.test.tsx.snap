// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test account group component: userSignupView. Renders correctly '<UserSignupView />' when content is loaded. 1`] = `
<section>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Styled(MuiBox)
      pb={10}
      pt={18}
    >
      <WithStyles(ForwardRef(Card))
        elevation={4}
      >
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-card-3"
        >
          <Styled(MuiBox)
            mb={3}
            textAlign="center"
          >
            <Memo(ForwardRef(AccountCircleIcon))
              className="makeStyles-account-1"
            />
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-caption-2"
            >
              Create a new account
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={2}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                sm={6}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="fname"
                  fullWidth={true}
                  id="firstName"
                  label="First name"
                  name="firstName"
                  onChange={[MockFunction]}
                  required={true}
                  value="Ester"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                sm={6}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="lname"
                  fullWidth={true}
                  id="lastName"
                  label="Last name"
                  name="lastName"
                  onChange={[MockFunction]}
                  required={true}
                  value="Exposito"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="email"
                  fullWidth={true}
                  id="email"
                  label="Email address"
                  name="email"
                  onChange={[MockFunction]}
                  required={true}
                  value="ester.exposito@gmail.com"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(TextField))
                  autoComplete="current-password"
                  fullWidth={true}
                  id="password"
                  label="Password"
                  name="password"
                  onChange={[MockFunction]}
                  required={true}
                  type="password"
                  value="madrilena123"
                  variant="outlined"
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={12}
              >
                <WithStyles(ForwardRef(FormControlLabel))
                  control={
                    <WithStyles(Component)
                      checked={false}
                      id="terms"
                      name="terms"
                      onChange={[MockFunction]}
                    />
                  }
                  label="I agree to the terms of use and privacy policy."
                />
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
            <Styled(MuiBox)
              my={2}
            >
              <ActiveButton />
            </Styled(MuiBox)>
            <Styled(MuiBox)
              textAlign="right"
            >
              <RedirectTo
                name="Already have an account? Sign in"
                path="/signin"
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
